src/EditorFeatures/TestUtilities/Completion/AbstractCompletionProviderTests.cs

```diff
@@ -33,20 +33,20 @@
 namespace Microsoft.CodeAnalysis.Editor.UnitTests.Completion
 {
     [UseExportProvider]
-    public abstract class AbstractCompletionProviderTests<TWorkspaceFixture> : TestBase, IClassFixture<TWorkspaceFixture>
+    public abstract class AbstractCompletionProviderTests<TWorkspaceFixture> : TestBase
         where TWorkspaceFixture : TestWorkspaceFixture, new()
     {
         private static readonly TestComposition s_baseComposition = EditorTestCompositions.EditorFeatures.AddExcludedPartTypes(typeof(CompletionProvider));
 
+        private readonly object _workspaceFixtureGate = new();
+        private ReferenceCountedDisposable<TWorkspaceFixture>.WeakReference _weakWorkspaceFixture;
```

Comment:
definitely worth comments.

Reply:
Extracted a helper type to hide complexity behind a simple API, and added comments to the implementation.